test/powershell/Language/Scripting/OutErrorVariable.Tests.ps1

```diff
@@ -0,0 +1,385 @@
+
+        $foo_err | Should Be "foo-error"
+
+        It '$get_item_err.count' { $get_item_err.count | Should Be 1 }
+        It '$get_item_err[0].exception' { $get_item_err[0].exception.GetType() | Should Be 'System.Management.Automation.ItemNotFoundException' }
+    }
+
+    It 'Multi-command pipeline with nested commands' {       
+
+        function get-bar3 
+        {
+            [CmdletBinding()]
+            param([Parameter(ValueFromPipeline = $true)][string] $i)
+
+            write-error  'bar-error'
+            write-output 'bar-output'
+            get-foo
+        }
+
+        (get-foo -errorVariable foo_err | get-bar3 -errorVariable bar_err) 2>&1 > $null        
+
+        $foo_err | Should be 'foo-error'
+        $bar_err | Should be @("bar-error", "foo-error")
+    }
+
+    It 'multi-command pipeline with nested commands' { 
+
+        function get-foo6
+        {
+            [CmdletBinding()]
+            param([Parameter(ValueFromPipeline = $true)][string] $i)
+
+            write-error  "foo-error"
+            write-output $i
+        }
+
+        function get-bar4
+        {
+            [CmdletBinding()]
+            param([Parameter(ValueFromPipeline = $true)][string] $i)
+
+            write-error  "bar-error"
+            get-foo6 "foo-output" -errorVariable script:foo_err1 | get-foo6 -errorVariable script:foo_err2 
+        }
+
+        get-bar4 -errorVariable script:bar_err 2>&1 > $null
+
+        $script:foo_err1 | Should be "foo-error"
+        $script:foo_err2 | Should be "foo-error"
+        $script:bar_err | Should be @("bar-error", "foo-error")        
+    }
+
+    It 'Nested output variables' {        
+        function get-foo7
+        {
+            [CmdletBinding()]
+            param([Parameter(ValueFromPipeline = $true)][string] $output)
+
+            $output
+            write-error "foo-error"
+        }
+
+        function get-bar5
+        {
+            [CmdletBinding()]
+            param()
+
+            "bar-output"
+            write-error  "bar-error"
+            get-foo7 "foo-output" -ev script:foo_err1 -ov script:foo_out1 | get-foo7 -ev script:foo_err2 -ov script:foo_out2
+            get-foo7 "foo-output" -ev script:foo_err3 -ov script:foo_out3 | get-foo7 -ev script:foo_err4 -ov script:foo_out4
+        }
+       
+
+        get-bar5 -ev script:bar_err -ov script:bar_out 2>&1 > $null
+
+        $script:foo_out1 | Should be "foo-output"
+        $script:foo_err1 | Should be "foo-error"
+
+        $script:foo_out2 | Should be "foo-output"
+        $script:foo_err2 | Should be "foo-error"
+
+        $script:foo_out3 | Should be "foo-output"
+        $script:foo_err3 | Should be "foo-error"
+
+        $script:foo_out4 | Should be "foo-output"
+        $script:foo_err4 | Should be "foo-error"
+                
+        $script:bar_out | Should be @("bar-output", "foo-output", "foo-output")
+        $script:bar_err | Should be @("bar-error", "foo-error", "foo-error")
+    }
+}
```

Comment:
added it.

---
In reply to: [78430403](https://github.com/Powershell/Powershell/pull/2160#discussion_r78430403) [](ancestors = 78430403)